zcash_primitives/src/zip32.rs

```diff
@@ -131,7 +131,7 @@ impl DiversifierKey {
     /// Returns the first index starting from j that generates a valid
     /// diversifier, along with the corresponding diversifier. Returns
     /// an error if the diversifier space is exhausted.
-    fn diversifier(&self, mut j: DiversifierIndex) -> Result<(DiversifierIndex, Diversifier), ()> {
+    pub fn diversifier(&self, mut j: DiversifierIndex) -> Result<(DiversifierIndex, Diversifier), ()> {
```

Comment:
This isn't the greatest interface, so I'm not entirely happy with making it public, but I guess we can still redesign it in a subsequent minor release (since we are nowhere near 1.0.0).